7_react/react1/week3/TodoList/src/App.tsx

```diff
@@ -0,0 +1,57 @@
+import "./App.css";
+import { useEffect, useState } from "react";
+import TodoForm from "./components/TodoForm";
+import TodoList from "./components/TodoList";
+import useSWR from "swr";
+
+export type Todo = {
+  id: number,
+  description : string,
+  deadline: string
+}
+export type  DeleteTodo = (id : number) => void;
+
+
+
+const url: string = 'https://gist.githubusercontent.com/benna100/391eee7a119b50bd2c5960ab51622532/raw';
+const fetcher = async() => {
+  try{
+    const response = await fetch(url);
+    const data = await response.json();
+    return data;
+  }catch(error){
+    console.log(error)
+  }
+}
+function App() {
+  const {data : initialTodo, error} = useSWR(url , fetcher);
+  
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState<boolean>(true);
```

Comment:
You can utilize the isLoading status directly from useSWR instead of creating a separate state for isLoading.